src/lib.rs

```diff
@@ -505,6 +534,63 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(1000_000)] // TODO weights
+		pub fn clear_abandoned_match(
+			origin: OriginFor<T>,
+			match_id: T::Hash
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let chess_match = match Self::chess_matches(match_id) {
+				Some(m) => m,
+				None => return Err(Error::<T>::NonExistentMatch.into()),
+			};
+
+			ensure!(
+				(chess_match.state == MatchState::OnGoing(NextMove::Whites)) | 
+				(chess_match.state == MatchState::OnGoing(NextMove::Blacks)), 
+				Error::<T>::MatchNotOnGoing
+			);
+			
+			let now = <frame_system::Pallet<T>>::block_number();
+			let diff = now - chess_match.last_move;
+			let abandoned: bool = match chess_match.style {
+				MatchStyle::Bullet => diff > T::BulletPeriod::get() * 10u32.into(),
+				MatchStyle::Blitz => diff > T::BlitzPeriod::get() * 10u32.into(),
+				MatchStyle::Rapid => diff > T::RapidPeriod::get() * 10u32.into(),
+				MatchStyle::Daily => diff > T::DailyPeriod::get() * 10u32.into(),
+			};
+
+			ensure!(abandoned, Error::<T>::MatchNotAbandoned);
```

Comment:
we need to have a clear distinction of two similar but separate concepts:
- expired move: player A didn't make their move within the time-to-move, so player B has the right to claim victory within a limited but sufficiently long period of time (e.g.: 10x time-to-move)
- abandoned match: player A didn't make their move within the time, and player B never claimed their victory within their rightful time window

previously, this code block was checking for an expired move, and the `ensure!` on line `565` only accounted for that.

now, all execution after line `565` only happens for abandoned matches.
that makes it impossible for player B to claim their victory due to expired move in the time window between `T::_Period::get()` and `T::_Period::get()*10`.

player B won't have the opportunity to claim the victory within this time window, and now they have to compete with third parties that are trying to get their incentive shares, which is unfair.

Reply:
on lines `558`-`563` we need to check for expired moves.

then, on the `else` conditional of line `580`, we check if the match is actually abandoned before calling `chess_match.clear_abandoned_bet(&winner, &who)`

Reply:
True, I shouldn't have removed the `expired` check. Let me add that back in